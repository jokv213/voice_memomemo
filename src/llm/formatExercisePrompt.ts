interface ExerciseData {
  exercise?: string;
  weight?: number;
  reps?: number;
  sets?: number;
  side?: string;
}

export function formatExercisePrompt(rawText: string, parsedData: ExerciseData): string {
  return `あなたは筋力トレーニングの専門家です。音声入力から得られたトレーニング記録を、正確で一貫性のある形式に整理してください。

音声入力: "${rawText}"

現在の解析結果:
- 種目: ${parsedData.exercise || '不明'}
- 重量: ${parsedData.weight ? `${parsedData.weight}kg` : '不明'}
- 回数: ${parsedData.reps ? `${parsedData.reps}回` : '不明'}
- セット数: ${parsedData.sets ? `${parsedData.sets}セット` : '不明'}
- 部位: ${parsedData.side || '不明'}

以下の形式で整理された記録を返してください:

種目: [正式な種目名]
重量: [数値]kg
回数: [数値]回
セット数: [数値]セット
部位: [左/右/両方/不明]

注意事項:
- 種目名は一般的な筋トレ用語を使用
- 数値は明確でない場合は「不明」
- 部位は「左」「右」「両方」「不明」のいずれか
- 一貫性を保ち、簡潔に記述`;
}
