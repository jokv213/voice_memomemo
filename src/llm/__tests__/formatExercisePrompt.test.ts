import {formatExercisePrompt} from '../formatExercisePrompt';

describe('formatExercisePrompt', () => {
  it('should generate prompt with complete exercise data', () => {
    const rawText = 'ベンチプレス100キロを10回3セット右側で実施';
    const parsedData = {
      exercise: 'ベンチプレス',
      weight: 100,
      reps: 10,
      sets: 3,
      side: '右',
    };

    const result = formatExercisePrompt(rawText, parsedData);

    expect(result).toMatchSnapshot();
    expect(result).toContain('ベンチプレス100キロを10回3セット右側で実施');
    expect(result).toContain('ベンチプレス');
    expect(result).toContain('100kg');
    expect(result).toContain('10回');
    expect(result).toContain('3セット');
    expect(result).toContain('右');
  });

  it('should generate prompt with partial exercise data', () => {
    const rawText = 'スクワットを何回かやった';
    const parsedData = {
      exercise: 'スクワット',
      weight: undefined,
      reps: undefined,
      sets: undefined,
      side: undefined,
    };

    const result = formatExercisePrompt(rawText, parsedData);

    expect(result).toMatchSnapshot();
    expect(result).toContain('スクワットを何回かやった');
    expect(result).toContain('スクワット');
    expect(result).toContain('不明');
  });

  it('should generate prompt with unknown exercise', () => {
    const rawText = '今日のトレーニングは良かった';
    const parsedData = {
      exercise: undefined,
      weight: undefined,
      reps: undefined,
      sets: undefined,
      side: undefined,
    };

    const result = formatExercisePrompt(rawText, parsedData);

    expect(result).toMatchSnapshot();
    expect(result).toContain('今日のトレーニングは良かった');
    expect(result).toContain('不明');
  });
});
