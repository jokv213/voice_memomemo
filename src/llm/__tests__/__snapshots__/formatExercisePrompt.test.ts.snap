// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatExercisePrompt should generate prompt with complete exercise data 1`] = `
"あなたは筋力トレーニングの専門家です。音声入力から得られたトレーニング記録を、正確で一貫性のある形式に整理してください。

音声入力: "ベンチプレス100キロを10回3セット右側で実施"

現在の解析結果:
- 種目: ベンチプレス
- 重量: 100kg
- 回数: 10回
- セット数: 3セット
- 部位: 右

以下の形式で整理された記録を返してください:

種目: [正式な種目名]
重量: [数値]kg
回数: [数値]回
セット数: [数値]セット
部位: [左/右/両方/不明]

注意事項:
- 種目名は一般的な筋トレ用語を使用
- 数値は明確でない場合は「不明」
- 部位は「左」「右」「両方」「不明」のいずれか
- 一貫性を保ち、簡潔に記述"
`;

exports[`formatExercisePrompt should generate prompt with partial exercise data 1`] = `
"あなたは筋力トレーニングの専門家です。音声入力から得られたトレーニング記録を、正確で一貫性のある形式に整理してください。

音声入力: "スクワットを何回かやった"

現在の解析結果:
- 種目: スクワット
- 重量: 不明
- 回数: 不明
- セット数: 不明
- 部位: 不明

以下の形式で整理された記録を返してください:

種目: [正式な種目名]
重量: [数値]kg
回数: [数値]回
セット数: [数値]セット
部位: [左/右/両方/不明]

注意事項:
- 種目名は一般的な筋トレ用語を使用
- 数値は明確でない場合は「不明」
- 部位は「左」「右」「両方」「不明」のいずれか
- 一貫性を保ち、簡潔に記述"
`;

exports[`formatExercisePrompt should generate prompt with unknown exercise 1`] = `
"あなたは筋力トレーニングの専門家です。音声入力から得られたトレーニング記録を、正確で一貫性のある形式に整理してください。

音声入力: "今日のトレーニングは良かった"

現在の解析結果:
- 種目: 不明
- 重量: 不明
- 回数: 不明
- セット数: 不明
- 部位: 不明

以下の形式で整理された記録を返してください:

種目: [正式な種目名]
重量: [数値]kg
回数: [数値]回
セット数: [数値]セット
部位: [左/右/両方/不明]

注意事項:
- 種目名は一般的な筋トレ用語を使用
- 数値は明確でない場合は「不明」
- 部位は「左」「右」「両方」「不明」のいずれか
- 一貫性を保ち、簡潔に記述"
`;
